File: python/PC.py
from Processor import Processor
from Disk import Disk
from Ram import Ram

class PC():

    # Constructor
    def __init__(self):
        self.procObj = Processor()
        self.diskObj = Disk()
        self.ramobj = Ram()
        self.totalPrice = self.procObj.getPrice() + self.diskObj.getPrice() + self.ramobj.getPrice()

    # Setter and Getter
    def setProcessor(self, procObj):
        self.procObj = procObj
    
    def getProcessor(self):
        return self.procObj
    
    def setDisk(self, diskObj):
        self.diskObj = diskObj
    
    def getDisk(self):
        return self.diskObj

    def setRam(self, ramObj):
        self.ramobj = ramObj
    
    def getRam(self):
        return self.ramobj

    def setTotalPrice(self):
        self.totalPrice = self.procObj.getPrice() + self.diskObj.getPrice() + self.ramobj.getPrice()
    
    def getTotalPrice(self):
        return self.totalPrice

    # Menampilkan atribut kelas
    def showPC(self):
        self.procObj.showProcessor()
        self.diskObj.showDisk()
        self.ramobj.showRam()
        print("Total Price     : Rp. " + str(self.totalPrice))

File: java/PC.java
public class PC {
    
    private int totalPrice;
    private Processor procObj;
    private Disk diskObj;
    private Ram ramObj;

    // Constructor
    PC(){
    }
    PC(Processor p, Disk d, Ram r){
        this.procObj = p;
        this.diskObj = d;
        this.ramObj = r;
        this.totalPrice = procObj.getPrice() + diskObj.getPrice() + ramObj.getPrice();
    }

    //Getter and Setter
    public int getTotalPrice() {
        return totalPrice;
    }

    public Ram getRam() {
        return ramObj;
    }

    public void setRam(Ram ramObj) {
        this.ramObj = ramObj;
    }

    public Disk getDisk() {
        return diskObj;
    }

    public void setDisk(Disk diskObj) {
        this.diskObj = diskObj;
    }

    public Processor getProcessor() {
        return procObj;
    }

    public void setProcessor(Processor procObj) {
        this.procObj = procObj;
    }

    public void setTotalPrice() {
        this.totalPrice = procObj.getPrice() + diskObj.getPrice() + ramObj.getPrice();
    }

    // Menampilkan atribut PC
    public void showPC() {
        procObj.showProcessor();
        diskObj.showDisk();
        ramObj.showRam();
        System.out.println("Total Price     : Rp. " + totalPrice);
    }

}
